mailet/standard/src/main/java/org/apache/james/transport/mailets/StripAttachment.java

```diff
@@ -63,156 +68,158 @@
+            throw new MailetException(String.format("Unknown remove parameter value '%s' waiting for '%s', '%s' or '%s'.", 
+                    removeAttachments, REMOVE_MATCHED, REMOVE_ALL, REMOVE_NONE));
         }
 
         if (directoryName != null) {
-            try {
-                FileUtils.forceMkdir(new File(directoryName));
-            } catch (Exception e) {
-                throw new MailetException("Could not create directory ["
-                        + directoryName + "].", e);
-            }
+            createDirectory();
         }
 
-        decodeFilename = getBooleanParameter(
-                getInitParameter(DECODE_FILENAME_PARAMETER_NAME),
-                decodeFilename);
-        if (getInitParameter(REPLACE_FILENAME_PATTERN_PARAMETER_NAME) != null) {
-            filenameReplacingPatterns = new PatternExtractor()
-                    .getPatternsFromString(getInitParameter(REPLACE_FILENAME_PATTERN_PARAMETER_NAME));
+        decodeFilename = getBooleanParameter(getInitParameter(DECODE_FILENAME_PARAMETER_NAME), DECODE_FILENAME_DEFAULT_VALUE);
+        String replaceFilenamePattern = getInitParameter(REPLACE_FILENAME_PATTERN_PARAMETER_NAME);
+        if (replaceFilenamePattern != null) {
+            filenameReplacingPatterns = new PatternExtractor().getPatternsFromString(replaceFilenamePattern);
+        } else {
+            filenameReplacingPatterns = ImmutableList.of();
         }
 
-        String toLog = String.format("StripAttachment is initialised with regex pattern [%s / %s]",
-                patternString, notpatternString);
-        if (directoryName != null) {
-            toLog += String.format(" and will save to directory [%s]", directoryName);
+        logConfiguration();
+    }
+
+    private Pattern regExFromParameter(String patternParameterName) throws MailetException {
+        String patternString = getInitParameter(patternParameterName);
+        try {
+            if (patternString != null) {
+                return Pattern.compile(patternString);
+            }
+            return null;
+        } catch (Exception e) {
+            throw new MailetException("Could not compile regex [" + patternString + "].");
         }
-        if (attributeName != null) {
-            toLog += String.format(" and will store attachments to attribute [%s]", attributeName);
+    }
+
+    private void createDirectory() throws MailetException {
+        try {
+            FileUtils.forceMkdir(new File(directoryName));
+        } catch (Exception e) {
+            throw new MailetException("Could not create directory [" + directoryName + "].", e);
         }
-        log(toLog);
     }
 
+      private void logConfiguration() {
+      StringBuilder logMessage = new StringBuilder();
+      logMessage.append("StripAttachment is initialised with regex pattern [");
+      if (regExPattern != null) {
+          logMessage.append(regExPattern.pattern());
+      } else {
+          logMessage.append("");
```

Comment:
what's the point of appending nothing? maybe you can simply remove these useless else clauses?